transcription/tasks.py

```diff
@@ -0,0 +1,145 @@
+        
+        # Retry the task if we haven't exceeded max retries
+        if self.request.retries < self.max_retries:
+            logger.info(f"Retrying task for SessionAudio ID {session_audio_id} (attempt {self.request.retries + 1})")
+            raise self.retry(exc=exc)
+        else:
+            logger.error(f"Max retries exceeded for SessionAudio ID {session_audio_id}")
+            return False
+
+
+@shared_task(bind=True, max_retries=2, default_retry_delay=30)
+def generate_session_log_task(self, gamelog_id, method="concat", model="gpt-4o"):
+    """
+    Generate a session log asynchronously.
+    
+    Args:
+        gamelog_id: ID of the GameLog instance
+        method: Method to use for log generation ('concat' or 'segment')
+        model: OpenAI model to use
+        
+    Returns:
+        str: Generated session log or None if failed
+    """
+    try:
+        logger.info(f"Starting session log generation for GameLog ID: {gamelog_id}")
+        
+        # Import here to avoid circular imports
+        from nucleus.models import GameLog
+        from .services import TranscriptionService
+        
+        try:
+            gamelog = GameLog.objects.get(id=gamelog_id)
+        except ObjectDoesNotExist:
+            logger.error(f"GameLog with ID {gamelog_id} not found")
+            return None
+            
+        # Generate the session log using the static method
+        result = TranscriptionService.generate_session_log_from_transcripts(
+            gamelog=gamelog,
+            model=model,
+            method=method
+        )
+        
+        logger.info(f"Successfully generated session log for GameLog ID: {gamelog_id}")
+        return result
+        
+    except Exception as exc:
+        logger.error(f"Error generating session log for GameLog ID {gamelog_id}: {str(exc)}")
+        
+        # Retry the task if we haven't exceeded max retries
+        if self.request.retries < self.max_retries:
+            logger.info(f"Retrying session log task for GameLog ID {gamelog_id} (attempt {self.request.retries + 1})")
+            raise self.retry(exc=exc)
+        else:
+            logger.error(f"Max retries exceeded for session log generation for GameLog ID {gamelog_id}")
+            return None
+
+
+@shared_task
+def cleanup_old_audio_files_task():
+    """
+    Cleanup old audio files and temporary files.
+    
+    This task can be scheduled to run periodically to clean up
+    temporary audio chunks and old processed files.
+    """
+    try:
+        logger.info("Starting cleanup of old audio files")
+        
+        # Import here to avoid circular imports
+        from .utils import cleanup_temporary_files
+        
+        cleanup_temporary_files()
+        
+        logger.info("Successfully completed audio files cleanup")
+        return True
+        
+    except Exception as exc:
+        logger.error(f"Error during audio files cleanup: {str(exc)}")
+        return False
+
+
+@shared_task
```

Comment:
This is a health check that's generic for celery. It isn't specific to the transcription app or functionality. The nucleus app is where I typically put code that is for the app generally.